src/lib.rs

```diff
@@ -64,38 +73,54 @@ pub trait Revisioned {
 	}
 }
 
+pub trait SerializeRevisioned {
+	/// Serializes the struct using the specficifed `writer`.
+	fn serialize_revisioned<W: Write>(&self, w: &mut W) -> Result<(), Error>;
+}
+
+pub trait DeserializeRevisioned {
+	/// Deserializes a new instance of the struct from the specficifed `reader`.
+	fn deserialize_revisioned<R: Read>(r: &mut R) -> Result<Self, Error>
+	where
+		Self: Sized;
+}
+
```

Comment:
Maybe require that both of these are subtraits of Revisioned?
```rust
trait SerializeRevisioned: Revisioned{ }
```

Reply:
Good idea, I was thinking about doing that. Updated.